x-pack/platform/plugins/private/file_upload/server/schemas.ts

```diff
@@ -26,26 +26,25 @@ export const analyzeFileQuerySchema = schema.object({
   timestamp_format: schema.maybe(schema.string()),
 });
 
-export const importFileQuerySchema = schema.object({
-  id: schema.maybe(schema.string()),
+const ingestPipeline = schema.object({
+  id: schema.string(),
+  pipeline: schema.maybe(schema.any()),
 });
 
-const ingestPipeline = schema.maybe(
-  schema.object({
-    id: schema.maybe(schema.string()),
-    pipeline: schema.maybe(schema.any()),
-  })
-);
-
-export const importFileBodySchema = schema.object({
+export const initializeImportFileBodySchema = schema.object({
   index: schema.string(),
-  data: schema.arrayOf(schema.any()),
+  /* Index settings */
   settings: schema.maybe(schema.any()),
   /** Mappings */
   mappings: schema.any(),
   /** Ingest pipeline definition */
-  ingestPipeline,
-  createPipelines: schema.maybe(schema.arrayOf(ingestPipeline)),
+  ingestPipelines: schema.arrayOf(ingestPipeline),
```

Comment:
Would ingestPipelines  ever be a `schema.maybe(schema.arrayOf(ingestPipeline))`? Or is it always empty array if there's no ingest pipeline?

Reply:
The latter. It should always be present and just empty if there are no pipelines to create.